BH/Modules/Item/Item.cpp

```diff
@@ -157,13 +163,372 @@ void GetItemStats()
+		{
+			flags |= ITEM_GROUP_POLEARM;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_CROSSBOW) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_CROSSBOW;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_JAVELIN) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_JAVELIN;  // TODO: this isn't being used. Fixme
+		}
+		if (ancestorTypes.find(ITEM_TYPE_THROWN) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_THROWING;
+		}
+
+		ItemAttributes* attrs = new ItemAttributes();
+		attrs->name = UnicodeToAnsi(GetTblEntryByIndex(pWeapons->wnamestr, TBLOFFSET_STRING));
+		attrs->code[0] = 0;
+		attrs->code[1] = 0;
+		attrs->code[2] = 0;
+		attrs->code[3] = 0;
+		attrs->code[4] = 0;
+		attrs->category = pWeapons->nType;
+		attrs->width = 0;
+		attrs->height = 0;
+		attrs->stackable = stackable;
+		attrs->useable = useable;
+		attrs->throwable = throwable;
+		attrs->itemLevel = 0;
+		attrs->unusedFlags = 0;
+		attrs->flags = flags;
+		attrs->flags2 = flags2;
+		attrs->qualityLevel = pWeapons->blevel;
+		attrs->magicLevel = pWeapons->bmagiclvl;
+		attrs->maxac = 0;
+		attrs->cost = 0;
+		ItemAttributeMap[GetTxtItemCode(pWeapons)] = attrs;
+	}
+}
+void GetArmorAttributes()
+{
+	D2ItemDataTbl* pItemDataTables = D2COMMON_10535_DATATBLS_GetItemDataTables();
+
+	for (auto d = 0; d < pItemDataTables->nArmorTxtRecordCount; d++)
+	{
+		ItemsTxt* pArmor = &pItemDataTables->pArmor[d];
+		BYTE stackable = pArmor->bstackable > 0 ? pArmor->bstackable : 0;
+		BYTE useable = pArmor->buseable > 0 ? pArmor->buseable : 0;
+		BYTE throwable = throwableMap[pArmor->nType] > 0 ? throwableMap[pArmor->nType] : 0; // Hey, you never know
+		unsigned int flags = ITEM_GROUP_ALLARMOR, flags2 = 0;
+
+		std::set<WORD> ancestorTypes;
+		FindAncestorTypes(pArmor->nType, ancestorTypes, parentMap1, parentMap2);
+
+		if (pArmor->dwcode == pArmor->dwultracode)
+		{
+			flags |= ITEM_GROUP_ELITE;
+		}
+		else if (pArmor->dwcode == pArmor->dwubercode)
+		{
+			flags |= ITEM_GROUP_EXCEPTIONAL;
+		}
+		else
+		{
+			flags |= ITEM_GROUP_NORMAL;
+		}
+
+		if (ancestorTypes.find(ITEM_TYPE_CIRCLET) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_CIRCLET;  // TODO: This kinda seems like it should be separate
+		}
+		else if (bodyLocMap[pArmor->nType] == EQUIP_HEAD)
+		{
+			flags |= ITEM_GROUP_HELM;
+		}
+		else if (bodyLocMap[pArmor->nType] == EQUIP_BODY)
+		{
+			flags |= ITEM_GROUP_ARMOR;
+		}
+		else if (bodyLocMap[pArmor->nType] == EQUIP_GLOVES)
+		{
+			flags |= ITEM_GROUP_GLOVES;
+		}
+		else if (bodyLocMap[pArmor->nType] == EQUIP_FEET)
+		{
+			flags |= ITEM_GROUP_BOOTS;
+		}
+		else if (bodyLocMap[pArmor->nType] == EQUIP_BELT)
+		{
+			flags |= ITEM_GROUP_BELT;
+		}
+		else if (bodyLocMap[pArmor->nType] == EQUIP_RIGHT_PRIMARY && ancestorTypes.find(ITEM_TYPE_ALLSHIELD) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_SHIELD;
+		}
+		flags = AssignClassFlags(pArmor->nType, ancestorTypes, flags);
+
+		ItemAttributes* attrs = new ItemAttributes();
```

Comment:
Just marking these as potential issues, needs the same treatment as the other comments